pkg/audit/manager.go

```diff
@@ -274,8 +275,63 @@ func (am *Manager) auditResources(
 	var errs opa.Errors
 	nsCache := newNSCache()
 
+	matchedKinds := make(map[string]bool)
+	if *auditMatchKindOnly {
+		constraintList := &unstructured.UnstructuredList{}
+	constraintsLoop:
+		for _, c := range constraintsGVK {
+			constraintList.SetGroupVersionKind(c)
+			if err = am.client.List(ctx, constraintList); err != nil {
+				am.log.Error(err, "Unable to list objects for gvk", "group", c.Group, "version", c.Version, "kind", c.Kind)
+				continue
+			}
+			for _, constraint := range constraintList.Items {
+				kinds, found, err := unstructured.NestedSlice(constraint.Object, "spec", "match", "kinds")
+				if err != nil {
+					am.log.Error(err, "Unable to return spec.match.kinds field", "group", c.Group, "version", c.Version, "kind", c.Kind)
+					// looking at all kinds if there is an error
+					matchedKinds["*"] = true
+					break constraintsLoop
+				}
+				if found {
+					for _, k := range kinds {
+						kind, ok := k.(map[string]interface{})
+						if !ok {
+							am.log.Error(errors.New("could not cast kind as map[string]"), "kind", k)
+							continue
+						}
+						kindsKind, _, err := unstructured.NestedSlice(kind, "kinds")
+						if err != nil {
+							am.log.Error(err, "Unable to return kinds.kinds field", "group", c.Group, "version", c.Version, "kind", c.Kind)
+							continue
+						}
+						for _, kk := range kindsKind {
+							if kk.(string) == "" {
+								// no need to continue, all kinds are included
+								matchedKinds["*"] = true
+								break constraintsLoop
+							}
+							// adding constraint match kind to matchedKinds list
+							matchedKinds[kk.(string)] = true
```

Comment:
Should we test for "*" here and break if `kk == "*"`?

Reply:
I don't think `*` is valid for kinds but added in case someone uses it

Reply:
Looks like we allow it in both places:

https://github.com/open-policy-agent/gatekeeper/blob/55962779f88a02ee575fd3e77b19cc2cebdbed60/pkg/target/regolib/src.rego#L137-L147